application/controllers/HostsController.php

```diff
@@ -38,6 +38,7 @@ public function indexAction()
         $db = $this->getDb();
 
         $hosts = Host::on($db)->with(['state', 'icon_image', 'state.last_comment']);
+        $hosts->getWith()['host.state']->setJoinType('INNER');
```

Comment:
Why not directly in the model?

Reply:
I can't give you a 100 percent answer here, but that can slow down other queries.